From 3af7da0eea7da9789f160599956b00471f25d6c1 Mon Sep 17 00:00:00 2001
From: Dan Petrisko <petrisko@cs.washington.edu>
Date: Sun, 20 Aug 2023 01:19:41 -0700
Subject: [PATCH] CONFIG

---
 config.sub | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/config.sub b/config.sub
index dba16e84c77..b68ebb92ae6 100755
--- a/config.sub
+++ b/config.sub
@@ -1754,7 +1754,7 @@ case $os in
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* )
+	     | fiwix* | dramfs*)
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
@@ -1792,6 +1792,8 @@ case $kernel-$os in
 		;;
 	*-eabi* | *-gnueabi*)
 		;;
+    elf-dramfs)
+        ;;
 	-*)
 		# Blank kernel with real OS is always fine.
 		;;
-- 
2.16.5

