From 496d94dd10362052a8fca0aca5d0cfae152e673a Mon Sep 17 00:00:00 2001
From: Dan Petrisko <petrisko@cs.washington.edu>
Date: Wed, 4 Sep 2024 13:38:27 -0700
Subject: [PATCH] CONFIG

---
 config.sub | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/config.sub b/config.sub
index defe52c0c87..b6200355900 100755
--- a/config.sub
+++ b/config.sub
@@ -1766,7 +1766,7 @@ case $os in
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* | mlibc* | cos* | mbr* )
+         | fiwix* | mlibc* | cos* | mbr* | dramfs* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
@@ -1859,6 +1859,8 @@ case $kernel-$os-$obj in
 		# None (no kernel, i.e. freestanding / bare metal),
 		# can be paired with an machine code file format
 		;;
+    elf-dramfs-)
+        ;;
 	-*-)
 		# Blank kernel with real OS is always fine.
 		;;
-- 
2.16.5

