From c81c7ef623adc65c624b41987ec9e45b7d365aff Mon Sep 17 00:00:00 2001
From: Dan Petrisko <petrisko@cs.washington.edu>
Date: Sun, 25 Aug 2024 21:01:07 -0700
Subject: [PATCH] CONFIG

---
 config.sub | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/config.sub b/config.sub
index 9a7a3ed642d..03a3f8becfd 100755
--- a/config.sub
+++ b/config.sub
@@ -1760,7 +1760,7 @@ case $os in
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* | mlibc* | cos* | mbr* )
+	     | fiwix* | mlibc* | cos* | mbr* | dramfs* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
@@ -1817,6 +1817,8 @@ case $kernel-$os in
 		;;
 	*-eabi* | *-gnueabi*)
 		;;
+    elf-dramfs-)
+        ;;
 	-*)
 		# Blank kernel with real OS is always fine.
 		;;
-- 
2.16.5

