From 2efec7c0e6ccf00a3d7a57584805452c032ef192 Mon Sep 17 00:00:00 2001
From: Dan Ruelas-Petrisko <petrisko@cs.washington.edu>
Date: Sun, 20 Jul 2025 13:39:41 -0700
Subject: [PATCH 1/3] Adding new encodings

---
 encoding.h | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/encoding.h b/encoding.h
index 01889d1..f90dcac 100644
--- a/encoding.h
+++ b/encoding.h
@@ -290,25 +290,39 @@
 #define IRQ_M_TIMER  7
 #define IRQ_U_EXT    8
 #define IRQ_S_EXT    9
 #define IRQ_VS_EXT   10
 #define IRQ_M_EXT    11
 #define IRQ_S_GEXT   12
 #define IRQ_COP      12
 #define IRQ_LCOF     13
 
 #define DEFAULT_RSTVEC     0x00001000
-#define CLINT_BASE         0x02000000
+#define CLINT_BASE         0x00300000
 #define CLINT_SIZE         0x000c0000
 #define EXT_IO_BASE        0x40000000
 #define DRAM_BASE          0x80000000
 
+#define UIO_BASE          0x200000000
+#define KIO_BASE          0x100000000
+#define MIO_BASE          0x000000000
+#define HOST_BASE          0x00100000
+#define CFG_BASE           0x00200000
+#define CLINT_BASE         0x00300000
+#define CACHE_BASE         0x00400000
+
+#define HOST_PUTCHAR       0x1000
+#define HOST_FINISH        0x2000
+
+#define CLINT_MTIME        0xbff8
+#define CLINT_MTIMECMP     0x4000
+
 /* page table entry (PTE) fields */
 #define PTE_V     0x001 /* Valid */
 #define PTE_R     0x002 /* Read */
 #define PTE_W     0x004 /* Write */
 #define PTE_X     0x008 /* Execute */
 #define PTE_U     0x010 /* User */
 #define PTE_G     0x020 /* Global */
 #define PTE_A     0x040 /* Accessed */
 #define PTE_D     0x080 /* Dirty */
 #define PTE_SOFT  0x300 /* Reserved for Software */
-- 
2.18.4

