From bb28a0d8dc05034f53d3a621f57f58c6666adfad Mon Sep 17 00:00:00 2001
From: Dan Petrisko <petrisko@cs.washington.edu>
Date: Mon, 23 Sep 2024 00:40:27 -0700
Subject: [PATCH 1/2] Fixing p macro

---
 p/riscv_test.h | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/p/riscv_test.h b/p/riscv_test.h
index 7bf35cf..0c80cad 100644
--- a/p/riscv_test.h
+++ b/p/riscv_test.h
@@ -198,9 +198,13 @@ handle_exception:                                                       \
         /* some unhandlable exception occurred */                       \
   1:    ori TESTNUM, TESTNUM, 1337;                                     \
   write_tohost:                                                         \
-        sw TESTNUM, tohost, t5;                                         \
-        sw zero, tohost + 4, t5;                                        \
-        j write_tohost;                                                 \
+        li a0, 0x00102000;                                              \
+        sw x0, (a0);                                                    \
+loop:                                                                   \
+        j loop;                                                         \
+  /*      sw TESTNUM, tohost, t5;  */                                   \
+  /*      sw zero, tohost + 4, t5; */                                   \
+  /*      j write_tohost;          */                                   \
 reset_vector:                                                           \
         INIT_XREG;                                                      \
         RISCV_MULTICORE_DISABLE;                                        \
-- 
2.16.5

