From 64247921ae943f25676b3ce4713634afe7b98948 Mon Sep 17 00:00:00 2001
From: Dan Petrisko <petrisko@cs.washington.edu>
Date: Mon, 23 Sep 2024 00:42:29 -0700
Subject: [PATCH 2/2] Fixing vm syscalls

---
 v/vm.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/v/vm.c b/v/vm.c
index 178d90b..f6118a8 100644
--- a/v/vm.c
+++ b/v/vm.c
@@ -40,7 +40,9 @@ static uint64_t lfsr63(uint64_t x)
 
 static void cputchar(int x)
 {
-  do_tohost(0x0101000000000000 | (unsigned char)x);
+  //do_tohost(0x0101000000000000 | (unsigned char)x);
+  uint8_t* putc_addr = (uint8_t*)(0x00101000);
+  *putc_addr = (char)x;
 }
 
 static void cputstring(const char* s)
@@ -51,7 +53,9 @@ static void cputstring(const char* s)
 
 static void terminate(int code)
 {
-  do_tohost(code);
+  //do_tohost(code);
+  uint8_t* finish_addr = (uint8_t*)(0x00102000);
+  *finish_addr = code;
   while (1);
 }
 
-- 
2.16.5

