From a7d74affef4f35489bb52443182d6f7c31a72952 Mon Sep 17 00:00:00 2001
From: Dan Ruelas-Petrisko <petrisko@cs.washington.edu>
Date: Wed, 6 Aug 2025 22:14:44 -0700
Subject: [PATCH] Disable lto

---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0a87335..bf7f218 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,25 +53,23 @@ add_compile_options(
         -std=c++11
         -g
         -Wall
         -Wno-parentheses
         -MMD
         -D_FILE_OFFSET_BITS=64
         -D_LARGEFILE_SOURCE
         -D_GNU_SOURCE
         -D__STDC_FORMAT_MACROS
         -march=native -mtune=native
-        -flto
 )
 
 add_link_options(
-        -flto
 )
 
 add_compile_options($<$<CONFIG:Release>:-Ofast>)
 
 if (TRACEOS)
     message(STATUS "TRACEOS is on.")
     add_compile_options(
             -DDUMP_UART
             -DDUMP_CLINT
             -DDUMP_PLIC
-- 
2.18.4

