From 2151e429853082cdd7b844a0acfa560bb03bd622 Mon Sep 17 00:00:00 2001
From: Dan Ruelas-Petrisko <petrisko@cs.washington.edu>
Date: Wed, 13 Aug 2025 22:31:30 -0700
Subject: [PATCH 2/2] Disable lto

---
 CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e2cd7e2..bf7f218 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,25 +53,23 @@ add_compile_options(
         -std=c++11
         -g
         -Wall
         -Wno-parentheses
         -MMD
         -D_FILE_OFFSET_BITS=64
         -D_LARGEFILE_SOURCE
         -D_GNU_SOURCE
         -D__STDC_FORMAT_MACROS
         -march=native -mtune=native
-        -flto
 )
 
 add_link_options(
-        -flto
 )
 
 add_compile_options($<$<CONFIG:Release>:-Ofast>)
 
 if (TRACEOS)
     message(STATUS "TRACEOS is on.")
     add_compile_options(
             -DDUMP_UART
             -DDUMP_CLINT
             -DDUMP_PLIC
@@ -79,20 +77,24 @@ if (TRACEOS)
             -DDUMP_ILLEGAL_INSTRUCTIONS
     )
 endif ()
 
 if (SIMPOINT)
     add_compile_options(
             -DSIMPOINT_BB
     )
 endif ()
 
+add_link_options(
+    -pthread
+)
+
 # Set Version Header
 set(CONFIG_VERSION "Dromajo-0.1")
 configure_file(include/config.h.in config.h @ONLY)
 
 if (WARMUP)
   message(STATUS "Checkpoint with WARMUP is on (typically for simpoints, but any checkpoint can use it).")
     add_compile_options( -DLIVECACHE)
 endif ()
 
 if (GOLDMEM)
-- 
2.18.4

