From 856b52d66e4881cc766eb7150edd1f71154d930e Mon Sep 17 00:00:00 2001
From: Dan Petrisko <petrisko@cs.washington.edu>
Date: Sun, 25 Aug 2024 20:58:56 -0700
Subject: [PATCH] CONFIG

---
 config.sub | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/config.sub b/config.sub
index 2c6a07ab3c3..9eec4fe3a55 100755
--- a/config.sub
+++ b/config.sub
@@ -1723,7 +1723,7 @@ fi
 
 case $os in
 	# Sometimes we do "kernel-libc", so those need to count as OSes.
-	llvm* | musl* | newlib* | relibc* | uclibc*)
+	llvm* | musl* | newlib* | relibc* | uclibc* | dramfs*)
 		;;
 	# Likewise for "kernel-abi"
 	eabi* | gnueabi*)
@@ -1866,6 +1866,8 @@ case $kernel-$os-$obj in
 		;;
 	*-eabi*- | *-gnueabi*-)
 		;;
+    elf-dramfs-)
+        ;;
 	none--*)
 		# None (no kernel, i.e. freestanding / bare metal),
 		# can be paired with an machine code file format
-- 
2.16.5

