diff --git a/benchmarks/common/test.ld b/benchmarks/common/test.ld
index a50b017..8331c7a 100644
--- a/benchmarks/common/test.ld
+++ b/benchmarks/common/test.ld
@@ -20,14 +20,13 @@ ENTRY(_start)
 
 SECTIONS
 {
+  . = 0x105000;
+  .tohost : { *(.tohost) }
 
   /* text: test code section */
   . = 0x80000000;
   .text.init : { *(.text.init) }
 
-  . = ALIGN(0x1000);
-  .tohost : { *(.tohost) }
-
   . = ALIGN(0x1000);
   .text : { *(.text) }
 
diff --git a/env b/env
--- a/env
+++ b/env
@@ -1 +1 @@
-Subproject commit 4fabfb4e0d3eacc1dc791da70e342e4b68ea7e46
+Subproject commit 4fabfb4e0d3eacc1dc791da70e342e4b68ea7e46-dirty
