From 310a4d249c5a8423b0924230007d4a38cd7ed675 Mon Sep 17 00:00:00 2001
From: Dan Petrisko <petrisko@cs.washington.edu>
Date: Mon, 26 Aug 2024 13:53:49 -0700
Subject: [PATCH] Updates to benchmarks

---
 benchmarks/common/syscalls.c | 27 +++++++++++++++++----------
 1 file changed, 17 insertions(+), 10 deletions(-)

diff --git a/benchmarks/common/syscalls.c b/benchmarks/common/syscalls.c
index 7a7b7fd..31c7c58 100644
--- a/benchmarks/common/syscalls.c
+++ b/benchmarks/common/syscalls.c
@@ -55,7 +55,9 @@ void setStats(int enable)
 
 void __attribute__((noreturn)) tohost_exit(uintptr_t code)
 {
-  tohost = (code << 1) | 1;
+  //tohost = (code << 1) | 1;
+  uint8_t* finish_addr = (uint8_t*)(0x00102000);
+  *finish_addr = code;
   while (1);
 }
 
@@ -76,7 +78,9 @@ void abort()
 
 void printstr(const char* s)
 {
-  syscall(SYS_write, 1, (uintptr_t)s, strlen(s));
+  //syscall(SYS_write, 1, (uintptr_t)s, strlen(s));
+  int i = 0;
+  while (s[i] != '\0') putchar(s[i]);
 }
 
 void __attribute__((weak)) thread_entry(int cid, int nc)
@@ -125,16 +129,19 @@ void _init(int cid, int nc)
 #undef putchar
 int putchar(int ch)
 {
-  static __thread char buf[64] __attribute__((aligned(64)));
-  static __thread int buflen = 0;
+  //static __thread char buf[64] __attribute__((aligned(64)));
+  //static __thread int buflen = 0;
 
-  buf[buflen++] = ch;
+  //buf[buflen++] = ch;
 
-  if (ch == '\n' || buflen == sizeof(buf))
-  {
-    syscall(SYS_write, 1, (uintptr_t)buf, buflen);
-    buflen = 0;
-  }
+  //if (ch == '\n' || buflen == sizeof(buf))
+  //{
+  //  syscall(SYS_write, 1, (uintptr_t)buf, buflen);
+  //  buflen = 0;
+  //}
+
+  uint8_t* putc_addr = (uint8_t*)(0x00101000);
+  *putc_addr = ch;
 
   return 0;
 }
-- 
2.16.5

