From aa02a56566005f9933443ec0e28d5f72b02974b4 Mon Sep 17 00:00:00 2001
From: Dan Ruelas-Petrisko <petrisko@cs.washington.edu>
Date: Sun, 20 Jul 2025 13:39:07 -0700
Subject: [PATCH 2/2] Fixing makefiles

---
 benchmarks/Makefile |  2 +-
 isa/Makefile        | 22 +++++-----------------
 2 files changed, 6 insertions(+), 18 deletions(-)

diff --git a/benchmarks/Makefile b/benchmarks/Makefile
index acc4573..63cc435 100644
--- a/benchmarks/Makefile
+++ b/benchmarks/Makefile
@@ -87,21 +87,21 @@ $(bmarks_riscv_out): %.riscv.out: %.riscv
 	$(RISCV_SIM) $< > $@
 
 riscv: $(bmarks_riscv_dump)
 run: $(bmarks_riscv_out)
 
 junk += $(bmarks_riscv_bin) $(bmarks_riscv_dump) $(bmarks_riscv_hex) $(bmarks_riscv_out)
 
 #------------------------------------------------------------
 # Default
 
-all: riscv
+all: $(bmarks_riscv_bin)
 
 #------------------------------------------------------------
 # Install
 
 date_suffix = $(shell date +%Y-%m-%d_%H-%M)
 install_dir = $(instbasedir)/$(instname)-$(date_suffix)
 latest_install = $(shell ls -1 -d $(instbasedir)/$(instname)* | tail -n 1)
 
 install:
 	mkdir $(install_dir)
diff --git a/isa/Makefile b/isa/Makefile
index e3b6719..ba3a165 100644
--- a/isa/Makefile
+++ b/isa/Makefile
@@ -81,61 +81,49 @@ $(1): $$($(1)_tests_dump)
 .PHONY: $(1)
 
 COMPILER_SUPPORTS_$(1) := $$(shell $$(RISCV_GCC) $(2) -c -x c /dev/null -o /dev/null 2> /dev/null; echo $$$$?)
 
 ifeq ($$(COMPILER_SUPPORTS_$(1)),0)
 tests += $$($(1)_tests)
 endif
 
 endef
 
-$(eval $(call compile_template,rv32ui,-march=rv32g -mabi=ilp32))
-$(eval $(call compile_template,rv32uc,-march=rv32g -mabi=ilp32))
-$(eval $(call compile_template,rv32um,-march=rv32g -mabi=ilp32))
-$(eval $(call compile_template,rv32ua,-march=rv32g -mabi=ilp32))
-$(eval $(call compile_template,rv32uf,-march=rv32g -mabi=ilp32))
-$(eval $(call compile_template,rv32ud,-march=rv32g -mabi=ilp32))
-$(eval $(call compile_template,rv32uzfh,-march=rv32g_zfh -mabi=ilp32))
-$(eval $(call compile_template,rv32uzba,-march=rv32g_zba -mabi=ilp32))
-$(eval $(call compile_template,rv32uzbb,-march=rv32g_zbb -mabi=ilp32))
-$(eval $(call compile_template,rv32uzbc,-march=rv32g_zbc -mabi=ilp32))
-$(eval $(call compile_template,rv32uzbs,-march=rv32g_zbs -mabi=ilp32))
-$(eval $(call compile_template,rv32si,-march=rv32g -mabi=ilp32))
-$(eval $(call compile_template,rv32mi,-march=rv32g -mabi=ilp32))
 ifeq ($(XLEN),64)
 $(eval $(call compile_template,rv64ui,-march=rv64g -mabi=lp64))
 $(eval $(call compile_template,rv64uc,-march=rv64g -mabi=lp64))
 $(eval $(call compile_template,rv64um,-march=rv64g -mabi=lp64))
 $(eval $(call compile_template,rv64ua,-march=rv64g -mabi=lp64))
 $(eval $(call compile_template,rv64uf,-march=rv64g -mabi=lp64))
 $(eval $(call compile_template,rv64ud,-march=rv64g -mabi=lp64))
-$(eval $(call compile_template,rv64uzfh,-march=rv64g_zfh -mabi=lp64))
 $(eval $(call compile_template,rv64uzba,-march=rv64g_zba -mabi=lp64))
 $(eval $(call compile_template,rv64uzbb,-march=rv64g_zbb -mabi=lp64))
 $(eval $(call compile_template,rv64uzbc,-march=rv64g_zbc -mabi=lp64))
 $(eval $(call compile_template,rv64uzbs,-march=rv64g_zbs -mabi=lp64))
 $(eval $(call compile_template,rv64mzicbo,-march=rv64g_zicboz -mabi=lp64))
 $(eval $(call compile_template,rv64si,-march=rv64g -mabi=lp64))
-$(eval $(call compile_template,rv64ssvnapot,-march=rv64g -mabi=lp64))
 $(eval $(call compile_template,rv64mi,-march=rv64g -mabi=lp64))
-$(eval $(call compile_template,hypervisor,-march=rv64gh -mabi=lp64))
 endif
 
 tests_dump = $(addsuffix .dump, $(tests))
 tests_hex = $(addsuffix .hex, $(tests))
 tests_out = $(addsuffix .out, $(filter rv64%,$(tests)))
 tests32_out = $(addsuffix .out32, $(filter rv32%,$(tests)))
+tests_riscv = $(addsuffix .riscv, $(filter rv64%,$(tests)))
 
 run: $(tests_out) $(tests32_out)
 
 junk += $(tests) $(tests_dump) $(tests_hex) $(tests_out) $(tests32_out)
+junk += $(tests_riscv)
 
 #------------------------------------------------------------
 # Default
 
-all: $(tests_dump)
+$(tests_riscv): %.riscv : %
+	@mv $< $@
+all: $(tests_riscv)
 
 #------------------------------------------------------------
 # Clean up
 
 clean:
 	rm -rf $(junk)
-- 
2.18.4

