From b0d5172b20c06f7895e4b233dfff247e73c77753 Mon Sep 17 00:00:00 2001
From: Dan Ruelas-Petrisko <petrisko@cs.washington.edu>
Date: Sun, 8 Feb 2026 07:52:25 -0800
Subject: [PATCH 2/2] Fixing makefiles

---
 benchmarks/Makefile | 3 ++-
 isa/Makefile        | 7 ++++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/benchmarks/Makefile b/benchmarks/Makefile
index 8c65579..11bf42d 100644
--- a/benchmarks/Makefile
+++ b/benchmarks/Makefile
@@ -87,21 +87,22 @@ $(bmarks_riscv_out): %.riscv.out: %.riscv
 	$(RISCV_SIM) $< > $@
 
 riscv: $(bmarks_riscv_dump)
 run: $(bmarks_riscv_out)
 
 junk += $(bmarks_riscv_bin) $(bmarks_riscv_dump) $(bmarks_riscv_hex) $(bmarks_riscv_out)
 
 #------------------------------------------------------------
 # Default
 
-all: riscv
+#all: 
+all: $(bmarks_riscv_bin)
 
 #------------------------------------------------------------
 # Install
 
 date_suffix = $(shell date +%Y-%m-%d_%H-%M)
 install_dir = $(instbasedir)/$(instname)-$(date_suffix)
 latest_install = $(shell ls -1 -d $(instbasedir)/$(instname)* | tail -n 1)
 
 install:
 	mkdir $(install_dir)
diff --git a/isa/Makefile b/isa/Makefile
index 0f0106f..e044f57 100644
--- a/isa/Makefile
+++ b/isa/Makefile
@@ -138,25 +138,30 @@ $(eval $(call compile_template,rv32uzbkb,-march=rv32g_zbkb -mabi=ilp32))
 $(eval $(call compile_template,rv32uzbkx,-march=rv32g_zbkx -mabi=ilp32))
 $(eval $(call compile_template,rv32uzbs,-march=rv32g_zbs -mabi=ilp32))
 $(eval $(call compile_template,rv32si,-march=rv32g -mabi=ilp32))
 $(eval $(call compile_template,rv32mi,-march=rv32g -mabi=ilp32))
 endif
 
 tests_dump = $(addsuffix .dump, $(tests))
 tests_hex = $(addsuffix .hex, $(tests))
 tests_out = $(addsuffix .out, $(filter rv64%,$(tests)))
 tests32_out = $(addsuffix .out32, $(filter rv32%,$(tests)))
+tests_riscv = $(addsuffix .riscv, $(filter rv64%,$(tests)))
 
 run: $(tests_out) $(tests32_out)
 
 junk += $(tests) $(tests_dump) $(tests_hex) $(tests_out) $(tests32_out)
+junk += $(tests_riscv)
 
 #------------------------------------------------------------
 # Default
 
-all: $(tests_dump)
+#all: $(tests_dump)
+$(tests_riscv): %.riscv : %
+	@mv $< $@
+all: $(tests_riscv)
 
 #------------------------------------------------------------
 # Clean up
 
 clean:
 	rm -rf $(junk)
-- 
2.18.4

