From: Dan Petrisko <petrisko@cs.washington.edu>
---

diff --git a/.gitmodules b/.gitmodules
index 6da2382..81afaaa 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -13,10 +13,14 @@
 	path = dejagnu
 	url = https://git.savannah.gnu.org/git/dejagnu.git
 	branch = master
+#[submodule "newlib"]
+#	path = newlib
+#	url = https://sourceware.org/git/newlib-cygwin.git
+#	branch = master
 [submodule "newlib"]
 	path = newlib
-	url = https://sourceware.org/git/newlib-cygwin.git
-	branch = master
+	url = https://github.com/bespoke-silicon-group/bsg_newlib_dramfs
+	branch = dramfs
 [submodule "gdb"]
 	path = gdb
 	url = https://sourceware.org/git/binutils-gdb.git
