From 65b0f30a53820731bf4e3126734a2a730b215533 Mon Sep 17 00:00:00 2001
From: Dan Petrisko <petrisko@cs.washington.edu>
Date: Tue, 27 Aug 2024 14:54:54 -0700
Subject: [PATCH] Changing newlib commit

---
 newlib | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/newlib b/newlib
index 26f7004..aa8b3d4 160000
--- a/newlib
+++ b/newlib
@@ -1 +1 @@
-Subproject commit 26f7004bf73c421c3fd5e5a6ccf470d05337b435
+Subproject commit aa8b3d4b13263c4c10d6d656ae91a7e81d48da7f
-- 
2.16.5

